homeassistant/components/devolo_home_network/button.py

```diff
@@ -0,0 +1,137 @@
+        icon="mdi:led-on",
+        name="Identify device with a blinking LED",
+        press_func=lambda device: device.plcnet.async_identify_device_start(),  # type: ignore[union-attr]
+    ),
+    PAIRING: DevoloButtonEntityDescription(
+        key=PAIRING,
+        icon="mdi:plus-network-outline",
+        name="Start PLC pairing",
+        press_func=lambda device: device.plcnet.async_pair_device(),  # type: ignore[union-attr]
+    ),
+    RESTART: DevoloButtonEntityDescription(
+        key=RESTART,
+        device_class=ButtonDeviceClass.RESTART,
+        entity_category=EntityCategory.CONFIG,
+        name="Restart device",
+        press_func=lambda device: device.device.async_restart(),  # type: ignore[union-attr]
+    ),
+    START_WPS: DevoloButtonEntityDescription(
+        key=START_WPS,
+        icon="mdi:wifi-plus",
+        name="Start WPS",
+        press_func=lambda device: device.device.async_start_wps(),  # type: ignore[union-attr]
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all devices and buttons and setup them via config entry."""
+    device: Device = hass.data[DOMAIN][entry.entry_id]["device"]
+
+    entities: list[DevoloButtonEntity] = []
+    if device.plcnet:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[IDENTIFY],
+                device,
+            )
+        )
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[PAIRING],
+                device,
+            )
+        )
+    if device.device and "restart" in device.device.features:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[RESTART],
+                device,
+            )
+        )
+    if device.device and "wifi1" in device.device.features:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[START_WPS],
+                device,
+            )
+        )
+    async_add_entities(entities)
+
+
+class DevoloButtonEntity(DevoloEntity, ButtonEntity):
+    """Representation of a devolo button."""
+
+    entity_description: DevoloButtonEntityDescription
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        description: DevoloButtonEntityDescription,
+        device: Device,
+    ) -> None:
+        """Initialize entity."""
+        self.entity_description = description
+        super().__init__(entry, device)
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        try:
+            await self.entity_description.press_func(self.device)
+        except DevicePasswordProtected as ex:
+            self.entry.async_start_reauth(self.hass)
+            raise HomeAssistantError(
+                f"Device {self.entry.title} is password protected"
```

Comment:
Is this the right error message? Seems like it's something like auth is expired or invalidated and needs to be reauthenticated?

Reply:
The devolo devices are a bit nasty here. As a user, you don't have to set a password and if you do you can still get data without using the password. On top, there is no non-writing endpoint I could check. So the first time Home Assistent notices a protected device is a failing action. 

That's why chose this message.

Reply:
Oh interesting. Let me see if I understand correctly, the flow is:
- set up the device in the config flow (either zero conf or user enters ip address). no password is set
- users presses button and the command fails, and starts reauth
- user enters password for the first time via the "reauth" flow

What exception is thrown if the password is incorrect? I'm assuming it's the same exception so maybe something like this:

`f"Device {self.entry.title} requires password or needs re-authenticatication"`

Reply:
I think this topic remains to be set.
I could also see a message something like `f"Device {self.entry.title} require re-authenticatication to set or change the password"`